(add) Talk Description and Speaker Info in /data/api/tracks.json

thisisayush opened the pull request:
- Added Description and Speaker Info to talks
- Converted Description from HTML to Markdown Text
- Added Scraping Script for generating /data/api/tracks.json from https://in.pycon.org/cfp/2017/proposals
- Added ```cfp``` key explicitly to all entries
- Added Missing Schedule (id:54)
- Fixed Missing/Wrong Schedule/Track Entries
#132

priyankt68 commented:
@thisisayush  Good work with scraping!

Some feedback :

1. Description is formatted twice in tracks.json. Please correct it.

2. For each div content with ```proposal-writeup--section``` class, remove the ```<h4 class="heading"><b>...</b></h4>```  tag. We don't need heading as "Description" in talk details section. This way we will have a completely formatted HTML. Currently it's not in ```tracks.json```.

Please comment back once you make these changes. Do let me know if you need any help :)

@rajataaron @CuriousLearner - Do comment if you guys have more to add.

Thank you.

CuriousLearner commented on data/api/tracks.json:
```diff
@@ -11,21 +11,25 @@
-			"description": "",
+			"title": "Deep Learning for NLP from scratch",
+			"description": "<br><h4 class=\"heading\"><b>Description:</b></h4><p><p><strong>King - Man + Woman = Queen</strong></p><p>The most famous example of word vectors paint an optimistic picture where computers can represent word into vectors which can be used to infer similarity. But can we extend it to sentences or to documents? How did word vectors come into existence? What are its utilities?</p ...
+			"date": "15 Jul, 2017",
```
Is the date of CFP used anywhere?

If not, we can avoid scraping it.

thisisayush commented:
Should I add New Line Character (\n) at the end of ```<h4>``` tags and at the start of every section?

realslimshanky commented:
@thisisayush that would be fine

thisisayush commented:
To conserve formatting and links within the description, I recommend storing description in the form of MARKDOWN text. This is recommended because:
- To Preserve links within text
- Automatic differentiation between Section Headings ```<h4>``` and description ```<p>``` text

thisisayush commented:
Updated Code and tracks.json. 
- Changed HTML to MARKDOWN Text to preserve formatting and links.
- Escaped ```\n``` to preserve line breaks
- Updated Missed Talk (id: 54 "Sponsored Workshop/Talks 4")
- Explicitly added ```cfp``` key to all entries in tracks.json file

priyankt68 commented:
@CuriousLearner  - I understand that it's good to keep plain text in json in order to use it across platforms. Do suggest on how to keep text formatting done by author if we are to avoid adding HTML in json.

In my opinion, it's fine to use HTML for website. For mobile, we would anyway need summarized description. Atleast, that's what I could understand from [this json](https://github.com/pythonindia/PyCon-Mobile-App/blob/master/eventsapp/data/jsonfiles/tracks.json) from [Pycon Mobile App](https://github.com/pythonindia/PyCon-Mobile-App) repository. (Though not actual tracks.json)

What do you think ? Please suggest.

priyankt68 commented:
@CuriousLearner Sounds good to me.

realslimshanky commented:
@CuriousLearner @thisisayush If we are talking about 2,3 lines we can chop off one paragraph and then add 'See more of this CFP' as link to CFP.
The same as the blog feed on the homepage of the website.
![image](https://user-images.githubusercontent.com/10004530/31319457-5d5576ba-ac81-11e7-9716-97b1c4e92dce.png)

realslimshanky commented:
@thisisayush it's better now to parse markdown to HTML using https://github.com/evilstreak/markdown-js . It has MIT license so we are free to use it.
Import the lib while in main.js and parse the markdown to HTML in line 378.
So the line will become `var description = markdown.toHTML( tracks[talk_id].description );`

thisisayush commented:
@realslimshanky The library you suggested was last updated 3-4 years back. I searched and found this library: Markdown-It (https://github.com/markdown-it/markdown-it), MIT license, and works flawlessly.
- Added Markdown to HTML Conversion in Talk Description.

realslimshanky commented on js/main.js:
```diff
@@ -375,7 +375,7 @@ jQuery(document).on('ready', function() {
+      var description = markdownit({breaks:true}).render(tracks[talk_id].description.replace(/\\n/g,"\n"));
       var speaker_name = tracks[talk_id].hasOwnProperty('speaker') ? tracks[talk_id].speaker.name : '';
       var time_duration = entity_details.start_time + ' - ' + entity_details.end_time;
       var display_title = speaker_name !== '' && typeof speaker_name !== 'undefined' ? title + ' by ' + speaker_name : title;
```
Delete this line.

thisisayush replied:
This line converts markdown text to HTML. Are you sure I shall remove this?

realslimshanky commented on js/main.js:
```diff
@@ -375,7 +375,7 @@ jQuery(document).on('ready', function() {
       var talk_id = schedule[i].talk_id;
       var entity_details = schedule[i];
       var title = entity_details.title;
```
change this to `var display_title = entity_details.title;` and delete line 381 since that line adds speaker name with the title which we have separately as `speaker_name`

thisisayush replied:
This issue has been resolved in d96d50910fbbd21ac5e189461eb1fba7ebf57b2b
